### DISCUSSION TITLE
Hugo Theming support

### DISCUSSION THREAD

**Discussion by ghost** [thuliteio/doks#73] [Q&A] [answered]
Hi!

Thanks for the work to put together this template and theme - it's really nice and minimal.

I'm considering potential use of this for the futue, but wanted to think about long-term theme updates and how that's easily integrated into existing, modified projects without creating huge conflicts.

I can't quite articulate the correct language here: But do you plan to introduce a different structure in the future? i.e. this becomes a submodule, which allows overriding layouts and therefore ~relatively easy theme updates? I.e. https://gohugobrasil.netlify.app/themes/customizing/

Thanks!

**Comment by h-enk** [comment, marked as answer]
Hi @badgerc, thanks for using Doks!

Like that you think long term 😉 

No, I'm not planning to release Doks as a submodule or module. I tried that, but Hugo does not support a npm based theme setup that way (yet). Apart from that, I don't like the parent-child relation with a submodule or module myself: when you have lots of customizations, a parent update might have no effect or even cause conflicts (I know that from experience). 

Doks can be seen as a "starter theme", meaning you build your site on a particular version of Doks. If you would like to update to a newer version of Doks, you'd have to do that manually by copying-pasting parts "intelligently". I know the latter sounds harsh, but is good doable (I know that from experience). Note also, that you're responsible for [updating dependencies](https://getdoks.org/docs/help/how-to-update/) yourself. I strongly recommend automating update notifications by using a [Dependabot workflow](https://github.com/h-enk/doks/blob/master/.github/dependabot.yml).

Hope that answers your question.
